### DISCUSSION TITLE
Modify OpenAPI Endpoints Individually

### DISCUSSION THREAD

**Discussion by ds5678** [dotnet/aspnetcore#57998] [Q&A] [answered]
I'd like to transform the OpenAPI document for each minimal api endpoint individually. There are operation transformers, but I haven't found a good way to choose whether or not to modify an operation. Currently, I just compare against the relevant endpoint paths.

Is there someway I could get the endpoint metadata when transforming operations? If I use `WithMetadata` on the relevant endpoints, the inserted metadata object could act as a flag for the operation transformation.

**Comment by martincostello** [comment, marked as answer]
See [here](https://github.com/martincostello/openapi-extensions/blob/dfb94dc5e3123268c7a090c1667ccd8094776f14/src/OpenApi.Extensions/Transformers/AddResponseDescriptionsTransformer.cs#L20) for an example of getting the endpoint metadata for an operation.

**Reply by ds5678** [reply]
I appreciate you answering. That is exactly what I was looking for.
